packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SurroundingPairScopeHandler/RangeLookupTree.ts

```diff
@@ -0,0 +1,73 @@
+import type { Range } from "@cursorless/common";
+import { RangeLookupList } from "./RangeLookupList";
+
+/**
+ * A tree of ranges that allows for efficient lookup of ranges that contain a search item.
+ * The items must be sorted in document order.
+ */
+export class RangeLookupTree<T extends { range: Range }> {
```

Comment:
As above, if correct, don't forget to rename doc.

```suggestion
/**
 * Given a list of ranges (the haystack), allows the client to search for smallest range containing a range (the needle).
 * Has the following requirements:
 * - the haystack must be sorted in document order
 * - **the needles must be in document order as well**. This enables us to avoid backtracking as you search for a sequence of items.
 * - the haystack entries **may** be nested, but one haystack entry cannot partially contain another
 */
export class OneWayNestedRangeFinder<T extends { range: Range }> {
```